src/antlr/index.ts

```diff
@@ -26,53 +25,49 @@ class SqlParseTree {
     ) => { table: { id: string; data: TableData }; columns: { id: string; data: ColumnData }[] } | undefined,
     mergedLeaves?: boolean
   ): Lineage<TableData, ColumnData> {
-    const lineageContext = new LineageContext(getTable, mergedLeaves ?? false);
-    const visitor = new LineageQueryVisitor<TableData, ColumnData>(lineageContext);
-    let lineage = this.tree.accept(visitor);
-    const outerRel = lineageContext.relationsStack.pop();
+    const visitor = new LineageVisitor<TableData, ColumnData>(getTable, mergedLeaves);
+    this.tree.accept(visitor);
+    const outerRel = visitor.relationsStack.pop();
     if (outerRel) {
-      lineage = visitor.aggregateResult(lineage, [outerRel.toLineage("[final result]")]);
+      visitor.onRelation(outerRel, "[final result]");
```

Comment:
For autocompletion I need the same "hack". I was wondering whether we could instead use another callback `onQuery` in addition to the existing ones. Or maybe split `onRelation` into two methods, one for when the relation is available and one for when there is also an alias. WDYT?

Reply:
Changed.